azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/VaultApi.java

```diff
@@ -0,0 +1,619 @@
+   @Path("/providers/Microsoft.KeyVault/deletedVaults")
+   @GET
+   @SelectJson("value")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<DeletedVault> listDeletedVaults();
+
+   @Named("vault:get_deleted")
+   @GET
+   @Path("/providers/Microsoft.KeyVault/locations/{location}/deletedVaults/{vaultName}")
+   @Fallback(NullOnNotFoundOr404.class)
+   DeletedVault getDeletedVault(@PathParam("location") String location, @PathParam("vaultName") String vaultName);
+
+   @Named("vault:purge")
+   @POST
+   @Fallback(FalseOnNotFoundOr404.class)
+   @Path("/providers/Microsoft.KeyVault/locations/{location}/deletedVaults/{vaultName}/purge")
+   boolean purgeVault(@PathParam("location") String location, @PathParam("vaultName") String vaultName);
+
+   // Key operations
+   @Named("key:list")
+   @SelectJson("value")
+   @GET
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   @Path("/keys")
+   @OAuthResource("https://vault.azure.net")
+   List<Key> listKeys(@EndpointParam URI keyVaultUri);
+
+   @Named("key:create")
+   @POST
+   @MapBinder(BindToJsonPayload.class)
+   @Path("/keys/{keyName}/create")
+   @OAuthResource("https://vault.azure.net")
+   KeyBundle createKey(@EndpointParam URI vaultBaseUrl, @PathParam("keyName") String keyName,
+         @Nullable @PayloadParam("attributes") KeyAttributes attributes,
+         @Nullable @PayloadParam("crv") String curveName, @Nullable @PayloadParam("key_ops") List<String> keyOps,
+         @PayloadParam("key_size") int keySize, @PayloadParam("kty") String keyType,
+         @Nullable @PayloadParam("tags") Map<String, String> tags);
+
+   @Named("key:import")
+   @PUT
+   @MapBinder(BindToJsonPayload.class)
+   @Path("/keys/{keyName}")
+   @OAuthResource("https://vault.azure.net")
+   KeyBundle importKey(@EndpointParam URI vaultBaseUrl, @PathParam("keyName") String keyName,
+         @PayloadParam("Hsm") boolean hsm, @Nullable @PayloadParam("attributes") KeyAttributes attributes,
+         @Nullable @PayloadParam("key") JsonWebKey key, @Nullable @PayloadParam("tags") Map<String, String> tags);
+
+   @Named("key:get")
+   @GET
+   @Path("/keys/{keyName}")
+   @Fallback(NullOnNotFoundOr404.class)
+   @OAuthResource("https://vault.azure.net")
+   KeyBundle getKey(@EndpointParam URI vaultBaseUrl, @PathParam("keyName") String keyName);
+
+   @Named("key:delete")
+   @DELETE
+   @Path("/keys/{keyName}")
+   @Fallback(NullOnNotFoundOr404.class)
+   @OAuthResource("https://vault.azure.net")
+   DeletedKeyBundle deleteKey(@EndpointParam URI vaultBaseUrl, @PathParam("keyName") String keyName);
+
+   @Named("key:get_versions")
+   @GET
+   @SelectJson("value")
+   @Path("/keys/{keyName}/versions")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   @OAuthResource("https://vault.azure.net")
+   List<Key> getKeyVersions(@EndpointParam URI vaultBaseUrl, @PathParam("keyName") String keyName);
+
+   @Named("key:update")
+   @PATCH
+   @MapBinder(BindToJsonPayload.class)
+   @Path("/keys/{keyName}{keyVersion}")
```

Comment:
Shouldn't this be `{keyName}/{keyVersion}`?

Reply:
I've seen the same thing in the certificates methods, and then the tests building the version parameter as `"/" + version`. Is that intentional?